### PR TITLE
Use selenium for web browsing

### PR DESCRIPTION
**Pull request by Agusx1211**
- Allows for browsing dynamic pages (Twitter, Google, .etc).
- It lets the agent read the whole page (paginated view)
- - Resuming can be added back later, but I think this performs better
- I added "No images" to the base prompt, otherwise it tries to fetch images.
- Selenium can later be used to further interact with the page (click buttons, fill forms).

### CODE REVIEW

**Comment by Durafen** [comment]
The AI appears to like paging and reading more detailed websites, however, I think there should be an option to obtain a get_text_summary of the whole webpage via prompt.

**Comment by Torantulino** [comment]
@Agusx1211 Could you please resubmit this as an optional (using an argument) rather than replacing the existing code?

There have been some good improvements to browsing since this was submitted, but we're still looking for a major solution.

**Comment by Agusx1211** [comment]
I can't work on this until late night. What do you mean as an optional? like adding a flag that changes the behavior?

**Comment by developerisnow** [comment]
> I can't work on this until late night. What do you mean as an optional? like adding a flag that changes the behavior?

Yes, looks like he meant this. Cause it's major improvement which needs separate flag

**Comment by ryanmac** [comment]
See also PR https://github.com/Torantulino/Auto-GPT/pull/96 which uses Playwright instead of requests.

But I also second the request that this is triggered by an argument.

**Comment by sberney** [comment]
I think this should be an alternate kind of command that the AI can run if the default (static) beautifulsoup parser fails.

**Comment by nponeccop** [comment]
An important thing is that Selenium makes it harder to setup. And it somehow assumes that Chrome is the only Selenium web driver. So, for it to merge I guess (I'm not the maintainer) it should be an entirely optional feature with a separate document describing the installation, keeping the current "for dummies" installation procedure as is.

**Review by nponeccop — CHANGES_REQUESTED** [review]
LGTM except the Chrome driver hardcoding. Support at least Firefox and Edge as well both in code and instructions.

And rebase against the current master.

File: scripts/data/prompt.txt
Comment (nponeccop) [inline scripts/data/prompt.txt:1]: Separate the prompt changes into a different PR.

**Review by nponeccop — CHANGES_REQUESTED** [review]
Resolve the conflicts and split the prompt changes.

**Comment by richbeales** [comment]
Selenium is now in the master branch

**Pull request closed without merging**